packages/hash/frontend/src/blocks/page/Comments/CommentBlock.tsx

```diff
@@ -57,16 +68,30 @@ export const ToggleTextExpandedButton: FunctionComponent<
 );
 
 type CommentProps = {
+  pageId: string;
   comment: PageComment;
 };
 
-export const CommentBlock: FunctionComponent<CommentProps> = ({ comment }) => {
-  const { hasText, author, textUpdatedAt } = comment;
+export const CommentBlock: FunctionComponent<CommentProps> = ({
+  pageId,
+  comment,
+}) => {
+  const { entityId, hasText, author, textUpdatedAt } = comment;
 
   const [collapsed, setCollapsed] = useState(true);
   const [shouldCollapse, setShouldCollapse] = useState(false);
+  const [editable, setEditable] = useState(false);
+  const [inputValue, setInputValue] = useState<TextToken[]>(hasText);
+
+  const { user } = useUser();
+  const [updateCommentText, { loading }] = useUpdateCommentText(pageId);
```

Comment:
Not important for this PR, but these changes are introducing application-specific context to the component. If/when we want to make this reusable elsewhere, we can pass down the `user` and `updateComment` function from outside the whole `CommentThread` (the same for `createComment`)